### DISCUSSION TITLE
PathLayer Custom Instanced Binary Attribute

### DISCUSSION THREAD

**Discussion by mcost45** [visgl/deck.gl#6140] [Q&A]
Hi again, I am trying to translate data objects individually (not entire layers). Perhaps there is a better approach, but:

To do this, I'm trying to add a custom binary attribute to the PathLayer - my aim is to supply a single x, y pair for each individual path group (so the total array length would be the size of startIndices x 2), so each path 'object' can be individually translated on the vertex shader. E.g. translations: [100, 100, 200, 200], for two path objects, would offset all of the vertices in the first path by (100, 100) and all of the vertices in the second path by (200, 200).

My vertex injection is already working if I use some arbitrary value instead of my attribute:

```
vs:#decl:
  attribute vec2 instanceTranslations;
vs:DECKGL_FILTER_GL_POSITION:
  float xOffset = project_size_to_pixel(instanceTranslations.x);
  float yOffset = project_size_to_pixel(instanceTranslations.y);
  position.xy += project_pixel_size_to_clipspace(vec2(xOffset, yOffset));
```
_(aside: is there a better way to convert a world dimension to clipspace?)_

My translations TypedArray attribute is being passed to the PathLayer. however, the attribute values passed in (instanceTranslations) to the vertex shader aren't working correctly. Is it possible for me to define an attribute in this way, or must I provide x y values for every single individual vertex (with the same repeated values for each path object)? For PathLayers, is each path object considered the 'instance'?

Currently I am using the attributeManager this way:
```
    attributeManager.addInstanced({
      instanceTranslations: {
        size: 2,
        type: GL.FLOAT,
        accessor: 'getTranslations',
        defaultValue: [0, 0]
      }
    });
```

**Comment by ilan-gold** [comment]
@mcost45 Potentially related to https://github.com/visgl/deck.gl/issues/6129.  Try upgrading deck.gl to the latest version (8.5.5, I believe)

**Reply by mcost45** [reply]
Thanks @ilan-gold - I checked out the issue and am on 8.5.5 already, but it seems like it could be a different issue (I want the divisor set as 1 for an instanced attribute).

To me it looks like PathLayer's individual 'instances' are not the path objects, but the individual lines that form each path (vertex-to-vertex segments) - or at least that's how my current instanced attribute is being passed through. Unless I am missing some attribute setting, it seems like this is expected - but I am still trying to find a good way to translate single path objects in parallel on the GPU (not the entire layer), instead of having to iterate through all the vertices and offset positions 🤔
